src/components/border_editor/border_editor.ts

```diff
@@ -0,0 +1,161 @@
+
+interface State {
+  menuState: MenuState;
+  activeTool: Tool;
+  borderColor: Color;
+  borderStyle: BorderStyle;
+  borderCommand: BorderCommand;
+}
+
+/**
+ * List the available borders positions and the corresponding icons.
+ * The structure of this array is defined to match the order/lines we want
+ * to display in the topbar's border tool.
+ */
+const BORDER_POSITIONS: [BorderCommand, string][][] = [
+  [
+    ["all", "o-spreadsheet-Icon.BORDERS"],
+    ["hv", "o-spreadsheet-Icon.BORDER_HV"],
+    ["h", "o-spreadsheet-Icon.BORDER_H"],
+    ["v", "o-spreadsheet-Icon.BORDER_V"],
+    ["external", "o-spreadsheet-Icon.BORDER_EXTERNAL"],
+  ],
+  [
+    ["left", "o-spreadsheet-Icon.BORDER_LEFT"],
+    ["top", "o-spreadsheet-Icon.BORDER_TOP"],
+    ["right", "o-spreadsheet-Icon.BORDER_RIGHT"],
+    ["bottom", "o-spreadsheet-Icon.BORDER_BOTTOM"],
+    ["clear", "o-spreadsheet-Icon.BORDER_CLEAR"],
+  ],
+];
+
+interface Props {}
+
+type BorderProperty =
+  | { name: "color"; value: Color }
+  | { name: "style"; value: BorderStyle }
+  | { name: "position"; value: BorderCommand };
+
+// -----------------------------------------------------------------------------
+// Border Editor
+// -----------------------------------------------------------------------------
+css/* scss */ `
+  .o-topbar-toolbar {
+    .o-toolbar-tools {
+      .o-border-dropdown {
+        padding: 4px;
+        display: flex;
+      }
+      .o-border-dropdown-section {
+        display: block;
+        .o-tool.o-border-color-tool {
+          padding: 1px 3px;
+        }
+        .o-tool.o-border-style-tool {
+          padding: 4px 3px;
+        }
+      }
+      .o-dropdown {
+        .o-dropdown-content {
+          .o-dropdown-line {
+            .o-line-item.active {
+              background-color: rgba(0, 0, 0, 0.2);
+            }
+          }
+        }
+      }
+    }
+  }
+  .o-style-preview {
+    margin-top: 8px;
+    margin-bottom: 6px;
+    width: 60px;
+  }
+  .o-style-thin {
+    border-bottom: 1px solid #000000;
+  }
+  .o-style-medium {
+    border-bottom: 2px solid #000000;
+  }
+  .o-style-thick {
+    border-bottom: 3px solid #000000;
+  }
+  .o-style-dashed {
+    border-bottom: 1px dashed #000000;
+  }
+  .o-style-dotted {
+    border-bottom: 1px dotted #000000;
+  }
+  .o-dropdown-border-type {
+    display: flex;
+  }
+  .o-dropdown-border-check {
+    width: 20px;
+    font-size: 16px;
+  }
+`;
+
+export class BorderEditor extends Component<Props, SpreadsheetChildEnv> {
+  static template = "o-spreadsheet-BorderEditor";
+  BORDER_POSITIONS = BORDER_POSITIONS;
+
+  static components = { ColorPicker };
+  borderStyles = borderStyles;
+  state: State = useState({
+    menuState: { isOpen: false, position: null, menuItems: [] },
+    activeTool: "",
+    borderStyle: DEFAULT_BORDER_DESC[0],
+    borderColor: DEFAULT_BORDER_DESC[1],
+    borderCommand: "clear",
+  });
+  openedEl: HTMLElement | null = null;
```

Comment:
blurk.
I know it's copy-pasted from the existing top bar, but it's not owly in any way.
And it's never used